packages/web/src/components/notification/Notification/TipReceivedNotification.tsx

```diff
@@ -29,18 +40,51 @@ const messages = {
 }
 
 type TipReceivedNotificationProps = {
-  notification: TipReceived
+  notification: TipReceive
+}
+
+/**
+ * Given a tip tx signature, fetches the corresponding reaction from DN,
+ * handles reaction writes, and optimistically updating local state.
+ */
+export const useReaction = (
+  tipTxSignature: string
+): [number, (val: number) => void] => {
+  const [discoveryReactionValue, setDiscoveryReactionValue] = useState(0)
+  const [localReactionValue, setLocalReactionValue] = useState<number | null>(
+    null
+  )
+
+  useEffect(() => {
+    const fn = async () => {
+      const reaction = await apiClient.getReaction({
+        reactedToIds: tipTxSignature
+      })
+      if (!reaction) return null
+      setDiscoveryReactionValue(reaction.reactionValue)
+    }
+    fn()
```

Comment:
nit: Maybe only call this if we don't already have a discovery value so it doesn't call discovery on every render? Or some other memoized technique

Reply:
discussed via slack